src/interfaces/vault.ts

```diff
@@ -304,6 +306,125 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     ).then(arr => arr.flat());
   }
 
+  /**
+   * Fetch the token amount that has been allowed to be used for deposits
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getDepositAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, tokenAddress, spenderAddress);
+  }
+
+  /**
+   * Fetch the token amount that has been allowed to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @returns TokenAllowance
+   */
+  async getWithdrawAllowance(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address
+  ): Promise<TokenAllowance> {
+    const spenderAddress = await this.getWithdrawSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.allowance(accountAddress, vaultAddress, spenderAddress);
+  }
+
+  /**
+   * Approve the token amount to allow to be used for deposits
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveDeposit(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    const spenderAddress = await this.getDepositSpenderAddress(vaultAddress, tokenAddress);
+    return this.yearn.tokens.approve(
+      accountAddress,
+      tokenAddress,
+      spenderAddress,
+      amount ?? MaxUint256.toString(),
+      overrides
+    );
+  }
+
+  /**
+   * Approve the token amount to allow to be used for withdraw
+   * @param accountAddress
+   * @param vaultAddress
+   * @param tokenAddress
+   * @param amount
+   * @param overrides
+   * @returns TransactionResponse
+   */
+  async approveWithdraw(
+    accountAddress: Address,
+    vaultAddress: Address,
+    tokenAddress: Address,
+    amount?: Integer,
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
```

Comment:
Is it not possible to use named params? With two optional params that would be great

Reply:
I would love to, but dont want to break the general pattern that exists through all the SDK using positional params. Think that change should be made in a major refactor.

Reply:
Or, we could stop using positional and start using named params :)

I'd prefer a more slower transition, I think it's ok to break a pattern if we want to stop using it

Reply:
I agree with @karelianpie, it's going to be hard to do a whole refactor to just do it one time. I think it would be better if we slowly move into named parameters. 

If anyone is using typescript (which they probably should) they are going to get the functions types and be able to check what type of parameters we are sending

Reply:
I do agree that new functions could start now to transition to named params as we do in v3 web repo too. We just need all to be aligned with these decision, and not something to be decided in this PR. We can talk about it with the rest :)